pkg/api/v1/types.go

```diff
@@ -710,6 +725,22 @@ type Container struct {
 	TTY   bool `json:"tty,omitempty" description:"Whether this container should allocate a TTY for itself, also requires 'stdin' to be true; default is false"`
 }
 
+func (ctr *Container) ApplyDefaults() {
```

Comment:
Some of our defaults are "same as that other field".  Maybe possible to do
that declaratively, but I expect corner cases.

On Tue, Sep 1, 2015 at 2:52 PM, Brendan Burns <notifications@github.com>
wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/13177#discussion_r38476943>
> :
>
> > @@ -710,6 +725,22 @@ type Container struct {
> >  	TTY   bool `json:"tty,omitempty" description:"Whether this container should allocate a TTY for itself, also requires 'stdin' to be true; default is false"`
> >  }
> >
> > +func (ctr *Container) ApplyDefaults() {
>
> To be clear:
>
> validation in code is great and appropriate, tags are probably not a good
> fit for the reasons you mentioned.
>
> defaulting in tags seems to make sense, it's self-documenting, and I don't
> think you should ever need code to do it.
>
> --brendan
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13177/files#r38476943>.
>